libs/grok/src/main/java/org/elasticsearch/grok/GrokCaptureExtracter.java

```diff
@@ -20,34 +20,55 @@
 /**
  * How to extract matches.
  */
-public abstract class GrokCaptureExtracter {
+public interface GrokCaptureExtracter {
+
     /**
      * Extract {@link Map} results. This implementation of {@link GrokCaptureExtracter}
      * is mutable and should be discarded after collecting a single result.
      */
-    static class MapExtracter extends GrokCaptureExtracter {
+    class MapExtracter implements GrokCaptureExtracter {
```

Comment:
Nit: I was wondering why you extracted this. I have a preference on how it was before because it was very clear in the file structure that this `MapExtracter` is related to `GrokCaptureExtracter` but I am curious to hear your thoughts. My apologies for not mentioning it earlier.

Reply:
Ignore.. you already fixed this...